### DISCUSSION TITLE
Some questions about TET element?

### DISCUSSION THREAD

**Discussion by QuatorzeM** [OpenRadioss/OpenRadioss#1738] [Q&A]
Dear Community,

As the discussions I have posted, I have been working on the TET10 element for a while.The code of openradioss inspires me a lot. Recently, I got some new difficulties to discuss :) 

1) For TET elements, especially TET10, the negative volume seems cannot be 100% avoided especially for low quality elements. Deleting the negative element or changing to the small strain case are two options provided in Openradioss. But in fact I doubt the efficiency of these two methods because for TET10. In my opinion, deleting an element can cause the geometric singularity which leads to the stress concentration and then the collapse of other elements. The small strain case can follow the same logic to cause the problem. So I'd like to know how do you think of these two methods ? 

2) In the theory book of Openradioss, I noticed the mass distribution is not unifome in all the nodes, 7/48 at the middile point，1/32 at the vertex (it should be calculated from 4 intergration points？). According to my tests, the mass distribution is important for the TET10 element stability. But I have noticed some code use 1/10 mass as each node and continue to use it in contact algorithms. In abaqus, it says 15 intergration point are used to calculate the mass distribution. I'm a little confused which one should be the most reasonable or how we should consider this question? 

3) Another issue is the contact calculation for TET10. In abaqus, instead of TET10,C3D10M has been proposed, the M signifies the modification for contact. I am not familiar with the contact algorithm, but I think the modification should be some special treatments for the 2nd order node of TET10. I'd like to know in openradioss, if there is any special treatment for TET10 contact algorithm or again, how we should consider this question? 

I hope these questions may interest you.
Best regards,

Fabien

**Comment by qiangRadioss** [comment]
Hi Fabien

For 1. i agree with you, deleting element option should be used very carefully, beside the possible issues you mentioned, contact issue could also be happened, for the switching to small strain option which is now frequently used in the "drop test " (e.g. phone) simulation, the geometry quality (or volume change) criteria are used for the switching, of course small strain option hasn't the same quality than large strain, but it's generally more robust and without time step issue, anyway with bad geometry, talking about quality is just luxury. 
If you have any new ideal about low quality element treatment, not hesitate to discuss.
For 2. i think the heterogeneous mass distribution  of T10 is only for the time step concern, once again, not hesitate to trying other solutions.
For 3. OpenRadioss uses sub-tria segments (w/ mid-nodes) for contact which is still linear. If you know how does C3D10M work, welcome to sharing with us?

Regards.
Qiang.

**Reply by QuatorzeM** [reply]
Hi Qiang,

Thanks again for your kind reply :) 

For small strain, as you said, we will try to do some test, to see its effect.  The TET element is still very crucial now. It seems the bad quality cannot be fully avoided especially we use lots of automatical tools to generate the mesh. I will the keep track of this problem to find some ideas, and to share. 

For 2, I will see the time step part of TET10 and try also the 15 intergration points in Abaqus to see its effect. 

For the contact, I think, in our case, if there is intersection (or penetration) at the beginning of simulation for TET10，the TET10 can easily collapse. The reason should be the less of treatement for this situation, we just delete the contact-pair if intersected . I am going to see if I can find some articles about the TET10 elements.

Recently, I will try to understand the part of mass scaling. Keep in touch.

Best wishes,

Fabien

**Reply by qiangRadioss** [reply]
Hi Fabien,

For your contact issue, which type of contact you used? 
mesh with initial intersection can't be used with type7, if type 24 or 25 is used, please look at flag Inacti in manual, Inacti=5 is recommended in general, and Inacti=-1 is used for high initial penetration cases.

Qiang.

**Reply by QuatorzeM** [reply]
Hi Qiang,

How are you doing ?

For your question, we are working on a contact algorithm named the defence node algorithm, not like penalty method, it's not widely used in commercial softwares. Generaly, it simplifies the contact problem as the collision of points, so that's why the mass distribution is important. I think the most difficult part in contact is the search of contact pair and the the initial intersection issues. The algorithm to calculate the contact force is not so difficult to implement.

I am recently working on the mass scaling of Openradioss ;)

Fabien

**Reply by qiangRadioss** [reply]
Hi Fabien,

right, your special contact algorithm is used, not the OpenRadioss'; interesting to know more as you said it's simpler than the penalty one.
I guess it's important to have the right contact pressure direction with your node-to-node (or node-to-nodes or nodes-to-nodes) algorithm, once the contact direction is the right one (not so easy neither using penalty method w/ node-to-segment or edge-to-edge), the intersection or initial penetration is not the big issue with solid model, the divergence due to high contact force could be managed by controlling either by stiffness or by penetration value, as we do with Inacti=-1,5 in type24.

Qiang.

**Reply by QuatorzeM** [reply]
Hi Qiang, 

My collegue told me for the defence node algorithm, the contact happens between a node and a segmentation. The force in the normal direction of the segmentation is calculated and then the friction force is calculated from the normal force with an coefficient. Is it the same in penalty method (I don't know much about the details)? For this method, we do have the problem of too high contact force, I will see the code in inacti=-1,5 in type24.

Thanks a lot.

**Reply by qiangRadioss** [reply]
Hi Fabien,

ok, looks the similar than penalty method.
For Inacti=5, we just shift the initial penetration for penetration compute (so that 0 penetration at beginning).

Qiang.
